File: src/main/java/at/fhhagenberg/sqe/entity/DoorState.java
package at.fhhagenberg.sqe.entity;

public enum DoorState {

    OPEN(1),
    CLOSED(2),
    OPENING(3),
    CLOSING(4);

    public final int doorState;

    DoorState(int doorState) {
        this.doorState = doorState;
    }

    public static DoorState valueOf(int doorState) {
        for (DoorState enumDoorState : values()) {
            if (enumDoorState.doorState == doorState) {
                return enumDoorState;
            }
        }
        return null;
    }
}

File: src/main/kotlin/at/fhhagenberg/sqe/entity/DoorState.kt
package at.fhhagenberg.sqe.entity

import at.fhhagenberg.sqe.util.EnumUtils

enum class DoorState(val doorState: Int) {
    OPEN(1), CLOSED(2), OPENING(3), CLOSING(4);

    companion object {
        @JvmStatic
        fun valueOf(doorState: Int): DoorState? = EnumUtils.valOf { it.doorState == doorState }
    }
}